pyuvdata/uvdata/mir_meta_data.py

```diff
@@ -0,0 +1,3087 @@
+            will set the values of `field_name` in the metadata array at the index
+            positions [0, 3]. Cannot be specified with `index` or `where`.
+        index : sequence of int
+            Optional argument, specifies the index positions at which to set the values
+            of the metadata. Cannot be specified with `header_key` or `where`.
+
+        Raises
+        ------
+        UserWarning
+            If attempting to set the field "dataoff", which is typically only used for
+            internal indexing purposes, and is generally not modified. Also raised if
+            modifying one of the known (pseudo) header keys.
+        """
+        if field_name == "dataoff":
+            warnings.warn(
+                'Values in "dataoff" are typically only used for internal indexing, '
+                "and should generally not be set by users. If you have set this in "
+                "error, you can undo this by using the reset method."
+            )
+        elif field_name in [
+            "inhid",
+            "blhid",
+            "sphid",
+            "ints",
+            "v_name",
+            "icode",
+            "antenna",
+            "achid",
+        ]:
+            warnings.warn(
+                "Changing fields that tie to header keys can result in unpredictable "
+                "behavior, and extreme care should be taken in directly modifying "
+                "them. If you have set this in error, you can undo this by using the "
+                "reset method."
+            )
+
+        idx_arr = self._index_query(use_mask, where, and_where_args, header_key, index)
+
+        # Make a copy of any changed variables, that we can revert if we ever
+        # happen to need to do so.
+        if field_name not in self._stored_values:
+            self._stored_values[field_name] = self._data[field_name].copy()
+
+        self._data[field_name][idx_arr] = value
+
+    def __setitem__(self, field_name, value):
+        """
+        Set values for a particular field using set_value.
+
+        field_name : str
+            Fields from which to extract data. Must match a field name, as list in the
+            `dtype` attribute of the object.
+        value : ndarray
+            Values to set the field in question to, where the provided selection
+            criteria match. Shape of the array must be broadcastable to either the shape
+            of the internal mask or to the shape of the `index` or `header_key`
+            arguments.
+        """
+        self.set_value(field_name=field_name, value=value)
+
+    def _generate_mask(
+        self,
+        where=None,
+        and_where_args=True,
+        header_key=None,
+        index=None,
+    ):
+        """
+        Generate a boolean mask based on selection criteria.
+
+        Note that this is an internal helper function which is not for general user use,
+        but instead is part of the low-level API for the MirMetaData object. Generates
```

Comment:
same comment about the "API" terminology.

Reply:
Fixed in a2f46b4b.

Reply:
Fixed in a2f46b4b.